### PR TITLE
Redeem offer

### PR DESCRIPTION
**Pull request by prabhaskumra**
A post request can be sent now to redeem an offer. After redemption the offer gets deleted from the database.

### CODE REVIEW

File: model/RedeemOffer.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    RedeemOffer: function RedeemOffer(offersAvailable, accountnumber){
+
+        db.read()
+        let offerData = []
+        offerData = db.get('offers').value()
+
+        let redeemOfferResultList = []
+        let transactionIdCount = db.get('transactionId').value()
+
+        var j = 0
```
Comment (daeeKang) [inline model/RedeemOffer.js]: I see where you are getting with this loop with the j variable. It makes sense to me but we also want to check if the offer couldn't be found in our offer pool. If that was the case we would return an error in the response status. To do that, you would have to rewrite your loop. 

I recommend using foreach function, foreach allows you to do something per item in an array. If you look at my RedeemComp.js or any of those, you can see how I use that. With that it's easier to read, and we can return an error message if we can't find the offer.

File: model/RedeemOffer.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                    var removed = offerData.splice(i,1)
+                    console.log(removed)
+                    db.set('offers', offerData).write()
+                    j++
+                    if(offersAvailable.length === j)
+                        break
+                }
+            }
+        }
+        var responseStatus = {"IsSuccess": true, "ErrorMessage": "", "ErrorCode": ""}
```
Comment (daeeKang) [inline model/RedeemOffer.js]: The offer would always return a hardcoded Success message but that shouldn't be the case if we couldn't find the offer

**Comment by prabhaskumra** [comment]
Made some changes. The responses are not hard code now.

**Comment by prabhaskumra** [comment]
Check out the changes I made...

**Pull request merged**